### PR TITLE
Feature/708 query endpoint http codes

### PR DESCRIPTION
**Pull request by HolgerReiseVSys**
# Changes

> Describe your changes in a short and concise list.

# Related issue

> Reference related issues, and use one of the [closing keywords, e.g. closes or fixes](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link the corresponding issue, if any.

# Additional information 

> Provide additional information for this change, if needed.

# Pre-Merge checklist

- [ ] New code is tested
- [ ] Present and new tests pass
- [ ] Documentation is updated
- [ ] The build is working without errors
- [ ] No new Sonar issues introduced
- [ ] Changelog is updated
- [ ] Code has been reviewed

### CODE REVIEW

File: rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/EhrController.java [hunk 1]
```diff
@@ -156,30 +157,38 @@ private EhrStatus extractEhrStatus(@RequestBody String content) {
...
+        try {
+            attributes = new ObjectMapper().readerForMapOf(Object.class).readValue(content);
+        } catch (JsonProcessingException e) {
+            throw new GeneralRequestProcessingException("Invalid content format", e);
+        }
 
-            if (atributes.containsKey(QUERYABLE)) {
-                ehrStatus.setQueryable((Boolean) atributes.get(QUERYABLE));
-            }
+        String subjectId = Optional.of("subjectId")
```
Comment (dmitrytulba) [inline rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/EhrController.java:171]: Please check Optional.of("subjectId") part

File: rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/EhrController.java [hunk 2]
```diff
@@ -156,30 +157,38 @@ private EhrStatus extractEhrStatus(@RequestBody String content) {
...
+                .map(attributes::get)
+                .map(String.class::cast)
+                .filter(StringUtils::isNotBlank)
+                .orElseThrow(() -> new InvalidApiParameterException("subjectId missing"));
+
+        String subjectNamespace = Optional.of("subjectNamespace")
```
Comment (dmitrytulba) [inline rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/EhrController.java:177]: Please check Optional.of("subjectNamespace") part

File: service/src/main/java/org/ehrbase/aql/compiler/AqlExpressionWithParameters.java [hunk 3]
```diff
@@ -50,17 +50,20 @@ public AqlExpressionWithParameters parse(String query, Map<String, Object> param
...
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
 
-        Map<String, Object> parameterMap = gson.fromJson(jsonParameterMap, Map.class);
+        Map<String, Object> parameters = (Map<String, Object>) parameterMap.get(PARAMETERS_KEY);
 
         // get the map from query-parameters
-        if (!parameterMap.containsKey(PARAMETERS_KEY))
+        if (parameters == null) {
```
Comment (dmitrytulba) [inline service/src/main/java/org/ehrbase/aql/compiler/AqlExpressionWithParameters.java:63]: Maybe it's good to check when there is no parames inside 

if (parameters == null || parameters.size() == 0) {

File: rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/EhrController.java (hunk 1 above)
Comment (HolgerReiseVSys) [reply rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/EhrController.java:171]: Do you mean if the literal is null?

File: service/src/main/java/org/ehrbase/aql/compiler/AqlExpressionWithParameters.java (hunk 3 above)
Comment (HolgerReiseVSys) [reply service/src/main/java/org/ehrbase/aql/compiler/AqlExpressionWithParameters.java:63]: It is OK if i's empty, but actually it looks like the code is not used at all...

**Review by dmitrytulba — APPROVED** [review]

**Pull request merged**